include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -133,17 +133,17 @@ __find_start_point(const _Rng1& __rng1, const _Index __rng1_from, _Index __rng1_
 // Do serial merge of the data from rng1 (starting from start1) and rng2 (starting from start2) and writing
 // to rng3 (starting from start3) in 'chunk' steps, but do not exceed the total size of the sequences (n1 and n2)
 template <typename _Rng1, typename _Rng2, typename _Rng3, typename _Index, typename _Compare>
-void
+std::pair<oneapi::dpl::__internal::__difference_t<_Rng1>, oneapi::dpl::__internal::__difference_t<_Rng2>>
 __serial_merge(const _Rng1& __rng1, const _Rng2& __rng2, _Rng3& __rng3, const _Index __start1, const _Index __start2,
-               const _Index __start3, const _Index __chunk, const _Index __n1, const _Index __n2, _Compare __comp)
+               const _Index __start3, const _Index __chunk, const _Index __n1, const _Index __n2, _Index __n3, _Compare __comp)
 {
     const _Index __rng1_size = std::min<_Index>(__n1 > __start1 ? __n1 - __start1 : _Index{0}, __chunk);
     const _Index __rng2_size = std::min<_Index>(__n2 > __start2 ? __n2 - __start2 : _Index{0}, __chunk);
     const _Index __rng3_size = std::min<_Index>(__rng1_size + __rng2_size, __chunk);
 
     const _Index __rng1_idx_end = __start1 + __rng1_size;
     const _Index __rng2_idx_end = __start2 + __rng2_size;
-    const _Index __rng3_idx_end = __start3 + __rng3_size;
+    const _Index __rng3_idx_end = std::min<_Index>(__n3, __start3 + __rng3_size);
```

Comment:
looks like a logical error, because `__n3` is the size but `__rng3_idx_end` is the last index.

Reply:
`__n3` also is the last index of range3.
Also as `__n1` is the last index  for range1 and `__n2` is the last index  for range2.